workers/src/main/scala/workers/IDBWorker.scala

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package workers
+
+import cats.effect.IO
+
+import scala.scalajs.js
+import org.scalajs.dom
+
+import js.annotation._
+
+import cats.effect.unsafe.implicits._
+
+/**
+  * Sample web worker, extremely simple just accept messages, prints them
+  * and answers a number
+  */
+@JSExportTopLevel("IDBWorker", moduleID = "worker")
+object IDBWorker {
```

Comment:
Neat.

We could define a `trait IOWebWorker` where you just have to define a `def onMessage(msg: dom.MessageEvent): IO[Unit]`.

Reply:
Right, this is still more on setting up the infrastructure, I thought about passing a partial function for message events too `PartialFunction[js.Any, IO[Unit]]`